src/plugins/metrics_sender.go

```diff
@@ -107,8 +107,23 @@ func (r *MetricsSender) Process(msg *core.Message) {
 
 func (r *MetricsSender) metricSenderBackoff(agentConfig *proto.AgentConfig) {
 	log.Debugf("update metric reporter client configuration to %+v", agentConfig)
-	backOffSettings := sdk.ConvertBackOffSettings(agentConfig.Details.Server.GetBackoff())
+
+	if agentConfig.GetDetails() == nil || agentConfig.GetDetails().GetServer() == nil || agentConfig.GetDetails().GetServer().GetBackoff() == nil {
```

Comment:
Hey, are we now no longer setting the defaults when the agentConfig Details etc are nil ?

Reply:
The agent is stateless, so it's up to the server to tell us what backoff settings they want. If you change the backoff settings and then want to update them again back to the default backoff setting, it is the server that needs to send us the default backoff settings again in the agent config command.